aten/src/ATen/TensorIterator.cpp

```diff
@@ -1302,7 +1302,12 @@ void TensorIteratorBase::set_output(int64_t output_idx, IntArrayRef sizes, IntAr
       // for the is_meta_ test.
       TORCH_INTERNAL_ASSERT(op.original_tensor.is_same(t));
       TORCH_INTERNAL_ASSERT(!op.tensor.is_same(t));
-      at::native::resize_output(op.tensor, sizes);
+      // fastpath CPU to skip a dispatcher trip
+      if (op.tensor.device().is_cpu()) {
+        at::native::resize_output_cpu(op.tensor, sizes);
+      } else {
+        at::native::resize_output(op.tensor, sizes);
+      }
```

Comment:
I put a note in the PR description: as more ops are ported to structured, fewer ops will actually hit either of these paths- the notable one in benchmarks was `at::eq`. Maybe I'll leave a comment here saying to delete the branching once more ops have been made structured?